src/interfaces.ts

```diff
@@ -41,8 +41,8 @@ export interface IFormProps<T = any> {
     initialModel?: any;
 }
 
-export interface IFormConfig<T = object, M = object> {
+export interface IFormConfig<T = any> {
     initialModel?: Partial<T>;
     onInputBlur?: (e: React.FocusEvent<any>) => any;
-    middleware?: (props: T) => T & M;
+    middleware?: (props: T) => any;
```

Comment:
Slight problem here. This reverts the issue raised in #27. If `middleware` returns anything other than an object this will break the library.

Reply:
Well spotted.
I think it was breaking for me. Let me double check.